In [86]:
import pandas as pd
from sklearn import preprocessing
from sklearn.metrics import accuracy_score, confusion_matrix
from sklearn.naive_bayes import GaussianNB
from sklearn.model_selection import train_test_split

In [87]:
data=pd.read_csv("train-1.csv")
features=[ 'Survived', 'Pclass', 'Sex', 'SibSp',
       'Parch', 'Embarked']
data.drop(["PassengerId","Name","Ticket","Fare","Cabin","Age"], axis=1,inplace=True)
data.isnull().sum()

Survived    0
Pclass      0
Sex         0
SibSp       0
Parch       0
Embarked    0
dtype: int64

In [88]:
label_encoder=preprocessing.LabelEncoder()
label_encoder.fit(data["Sex"])
data["Sex"]=label_encoder.transform(data["Sex"])
label_encoder.fit(data["Embarked"])
data["Embarked"]=label_encoder.transform(data["Embarked"])
data.head()

Survived  Pclass  Sex  SibSp  Parch  Embarked
0         0       3    1      1      0         2
1         1       1    0      1      0         0
2         1       3    0      0      0         2
3         1       1    0      1      0         2
4         0       3    1      0      0         2

In [83]:
for feat in features:
  y=data[feat]
  x=data.drop(feat,axis=1)
  x_train,x_test,y_train,y_test=train_test_split(x,y,random_state=1,test_size=0.25)
  clf=GaussianNB()
  y_pred=clf.fit(x_train,y_train).predict(x_test)
  print("Accuracy Score for {a} is {b}".format(a=feat,b=accuracy_score(y_test,y_pred,normalize=True)))
  print("Confusion Matrix for {a} is \n {b}".format(a=feat, b= confusion_matrix(y_test,y_pred)))
  print("\n")

Accuracy Score for Survived is 0.8071748878923767
Confusion Matrix for Survived is 
 [[112  26]
 [ 17  68]]


Accuracy Score for Pclass is 0.5022421524663677
Confusion Matrix for Pclass is 
 [[20 24 20]
 [ 6 15 23]
 [21 17 77]]


Accuracy Score for Sex is 0.7488789237668162
Confusion Matrix for Sex is 
 [[ 50  30]
 [ 26 117]]


Accuracy Score for SibSp is 0.13452914798206278
Confusion Matrix for SibSp is 
 [[ 21  12   0 104  13   0   0]
 [  6   3   0  31   6   0   0]
 [  1   3   0   4   1   0   1]
 [  0   0   0   1   3   0   2]
 [  0   0   0   0   4   0   3]
 [  0   0   0   0   0   0   3]
 [  0   0   0   0   0   0   1]]


Accuracy Score for Parch is 0.336322869955157
Confusion Matrix for Parch is 
 [[71  2  0 29 48 18  1]
 [ 8  0  2 15  6  3  0]
 [ 3  0  4  3  6  1  0]
 [ 0  0  0  0  1  0  0]
 [ 1  0  0  0  0  0  0]
 [ 0  0  0  0  1  0  0]
 [ 0  0  0  0  0  0  0]]


Accuracy Score for Embarked is 0.600896860986547
Confusion Matrix for Embarked is 
 [[ 13   3  29]
 [  0   6  12]
 [ 34  

#INFERENCE:
The accuracy score taking different categorical features into consideration is as follows:

SURVIVED:0.807
ie, 80.7%  of the records (180) are correcty classified, hence this model is about 80% accurate in classification.

PCLASS:0.5022
ie, 50.22%  of the records (112)are correcly classified, hence this model's accuracy is average.

SEX:.7422
ie, about 74.22% of the records (167) are classified correctly and hence it has a moderate accuracy rate.

SIBSP:0.1345
ie, only 13.45% of the records(30) are classified correctly hence it has a poor accuracy.

PARCH:0.3363
ie, 33.63% of the records (21) taking parch as dependent variable is correctly classfied hence is a poor model for classification.


EMBARKED:0.6008
ie, 60% of the records(134) are correcly classified and hence is average in accuracy.


***Hence the best model for classification is on the basis of survival attribute, with an accuracy rate of 80%.***